app/flatpak-builtins-install.c

```diff
@@ -300,10 +301,30 @@ flatpak_builtin_install (int argc, char **argv, GCancellable *cancellable, GErro
   if (!flatpak_split_partial_ref_arg (name, &opt_arch, &branch, error))
     return FALSE;
 
+  /* Make sure the default branch is pulled from the server's summary
+     file if it's defined there and hasn't been pulled in yet. */
+  default_branch = flatpak_dir_get_remote_default_branch (dir, repository);
```

Comment:
You read my mind :-). The initial use case we thought of was actually to provide a way for GNOME Software to filter out versions of the same application available from other branches, and for that having the default branch locally stored would be needed.

However, it's true that it seems a bit silly to ping the remote server every time an app is installed to get the default branch if it has not been retrieved yet, but that's the initial idea I came up with that would allow us to "push" it from the server to clients that have been already configured and have no default branch locally defined yet.

To be honest I also wonder whether this is the right and only place we should do this, maybe we should do it for every command that contacts the remote server (e.g. update, remote-ls...)?

But at that point I think I'd rather remove the fetching part from here and provide a new built-in command that would update the local information we have for the configured remotes with the one from the server's summary file (i.g. `flatpak remote-update`?), expose it via the public API and then maybe call it from GNOME Software on startup?

@alexlarsson What do you think?